### PR TITLE
types/data: the bit data type should be treat as uint64

### PR DESCRIPTION
**Pull request by gaoxingliang**
### What problem does this PR solve? 

Fix https://github.com/pingcap/tidb/issues/11895
the -1 can't be insert into a bit column.
### What is changed and how it works?
1. when convert to mysql bit, mapped it to unsigned int.
2. the error is wrong (it should be data too long instead of out of range)

### Check List 

Tests 

 - Unit test  -- added test cases

### CODE REVIEW

File: types/datum.go [hunk 1]
```diff
@@ -1214,12 +1214,17 @@ func (d *Datum) convertToMysqlBit(sc *stmtctx.StatementContext, target *FieldTyp
 	switch d.k {
 	case KindString, KindBytes:
 		uintValue, err = BinaryLiteral(d.b).ToInt(sc)
+	case KindInt64:
+		// if input kind is int64 (signed), when trans to bit, we need to treat it as unsigned
+		d.k = KindUint64
+		uintDatum, err1 := d.convertToUint(sc, target)
```
Comment (SunRunAway) [inline types/datum.go]: How about using `fallthrough` here?

Comment (gaoxingliang) [reply types/datum.go]: fixed

File: executor/insert_test.go [hunk 2]
```diff
@@ -15,6 +15,7 @@ package executor_test
 
 import (
 	"fmt"
+	"github.com/pingcap/tidb/types"
```
Comment (qw4990) [inline executor/insert_test.go]: Please format this statement.

Comment (gaoxingliang) [reply executor/insert_test.go]: fixed but the unit test failed without any error.

**Comment by gaoxingliang** [comment]
Can anyone help to review this PR?

**Comment by zz-jason** [comment]
> Can anyone help to review this PR?

Sorry for the late reply. @SunRunAway @qw4990 PTAL

**Pull request merged**